From 455ade716e2b89fe37eb32be09e168e3de371b6e Mon Sep 17 00:00:00 2001
From: Xianzhong <xianzhong.li@nxp.com>
Date: Fri, 1 Jun 2018 18:05:54 +0800
Subject: [PATCH 3899/5242] MGS-3947 gpu: imx: dpu-blit: fix wayland vg3d
 stuck on 8qxp

commit  880fb9b4c4800ae71badd54b2131e82ffe0efdf8 from
https://source.codeaurora.org/external/imx/linux-imx.git

gpu tiling is not enabled in vg3d, need bypass dprc/prg process,
dprc stuck since no pipeline sync when switch linear to tile.
add force sync for command sequence pipeline when switch to tile,
remove the frequent dprc config for the contigous linear blit.

Signed-off-by: Xianzhong <xianzhong.li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dpu-blit/dpu-blit.c |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/imx/dpu-blit/dpu-blit.c b/drivers/gpu/imx/dpu-blit/dpu-blit.c
index 7df60e0..b996502 100644
--- a/drivers/gpu/imx/dpu-blit/dpu-blit.c
+++ b/drivers/gpu/imx/dpu-blit/dpu-blit.c
@@ -119,7 +119,7 @@ void dpu_be_configure_prefetch(struct dpu_bliteng *dpu_be,
 	 * Waiting for the previous tiled command finished
 	 * before disable the dpr.
 	 */
-	if (tiled_work_unfinished) {
+	if (tiled_work_unfinished || baddr) {
 		dpu_be_wait(dpu_be);
 		dpu_cs_wait_idle(dpu_be);
 		udelay(10);
@@ -127,7 +127,8 @@ void dpu_be_configure_prefetch(struct dpu_bliteng *dpu_be,
 	}
 
 	if (baddr == 0x0) {
-		dprc_disable(dprc);
+		if (!start)
+			dprc_disable(dprc);
 		start = true;
 		return;
 	}
-- 
1.7.9.5

